mlflow/models/model.py

```diff
@@ -698,8 +698,17 @@ def log(
                 model_config = kwargs["model_config"]
                 if isinstance(model_config, str):
                     try:
-                        with open(model_config) as f:
-                            model_config = json.load(f)
+                        file_extension = os.path.splitext(model_config)[1].lower()
+                        if file_extension == ".json":
+                            with open(model_config) as f:
+                                model_config = json.load(f)
+                        elif file_extension in [".yaml", ".yml"]:
+                            with open(model_config) as f:
+                                model_config = yaml.safe_load(f)
+                        else:
+                            _logger.warning(
+                                "Unsupported file format for model config: %s", model_config
+                            )
```

Comment:
```suggestion
                            _logger.warning(
                                "Unsupported file format for model config: %s. Cannot log model config as params.", model_config
                            )
```

Can we modify this warning (and also the warning in the exception) to be something like this - and then do some validation in the next try that model_config is a dictionary, so that the warning with `'str' object has no attribute 'items'` isn't surfaced to users?

Reply:
The issue is coming from the fact that we tried to load the yaml file as json. So the fallback of str to reading as a dict won't work as well. So throwing this working makes sense IMO. what do you think?

Reply:
Yes - but if someone gives an unsupported file format it'll warn and then continue to also pass that file path to be logged as param? and give `'str' object has no attribute 'items'`. Although I believe there should be validations that the config file can be loaded before this point.

Reply:
Not super important but just seems like there will be multiple warnings for the same issue if there is unsupported format or loading the file fails - it will still try to log params and then warn again.